lib/compiler/aro_translate_c.zig

```diff
@@ -496,16 +711,125 @@ fn transType(c: *Context, scope: *Scope, raw_ty: Type, source_loc: TokenIndex) T
         .long_double => return ZigTag.type.create(c.arena, "c_longdouble"),
         .float80 => return ZigTag.type.create(c.arena, "f80"),
         .float128 => return ZigTag.type.create(c.arena, "f128"),
+        .@"enum" => @panic("TODO"),
+        .pointer,
+        .unspecified_variable_len_array,
+        .array,
+        .static_array,
+        .incomplete_array,
+        => @panic("TODO"),
         .func,
         .var_args_func,
         .old_style_func,
-        => return transFnType(c, scope, raw_ty, ty, source_loc, .{}),
+        => return transFnType(c, scope, ty, ty, source_loc, .{}),
+        .@"struct",
+        .@"union",
+        => {
+            var trans_scope = scope;
+            if (ty.isAnonymousRecord(c.comp)) {
+                const record_decl = ty.data.record;
+                const name_id = c.mapper.lookup(record_decl.name);
+                if (c.weak_global_names.contains(name_id)) trans_scope = &c.global_scope.base;
+            }
+            const name = c.decl_table.get(@intFromPtr(ty.data.record)).?;
+            return ZigTag.identifier.create(c.arena, name);
+        },
+        .attributed,
+        .typeof_type,
+        .typeof_expr,
+        => unreachable,
         else => return error.UnsupportedType,
     }
 }
 
-fn zigAlignment(bit_alignment: u29) u32 {
-    return bit_alignment / 8;
+/// This function returns a ?c_uint to match Clang's behaviour of using c_uint.
+/// This can be changed to a u29 after the Clang frontend for translate-c is removed.
+fn alignmentForField(
+    record_ty: Type,
+    head_field_alignment: ?c_uint,
+    field_index: usize,
+) ?c_uint {
+    const record_decl = record_ty.getRecord().?;
```

Comment:
Sorry - one other thing I just noticed - you can just pass the record decl in directly instead of having to call `getRecord` each time in this function - it looks like `record_ty` isn't used for anything else.

Reply:
Oh that's a good catch! Updating it now.